data/src/main/java/io/gnosis/data/repositories/UnstoppableDomainsRepository.kt

```diff
@@ -0,0 +1,38 @@
+package io.gnosis.data.repositories
+
+import com.unstoppabledomains.config.network.model.Network
+import com.unstoppabledomains.exceptions.ns.NamingServiceException
+import com.unstoppabledomains.resolution.DomainResolution
+import com.unstoppabledomains.resolution.Resolution
+import com.unstoppabledomains.resolution.naming.service.NamingServiceType
+import io.gnosis.data.BuildConfig
+import pm.gnosis.model.Solidity
+import pm.gnosis.utils.asEthereumAddress
+import java.util.concurrent.Callable
+import java.util.concurrent.FutureTask
+
+class BackgroundTask : Callable<String> {
+    var address = ""
+    var domain: String;
+
+    constructor(domain: String)  {
+        this.domain = domain;
+    }
+
+    override fun call(): String {
+        val library: DomainResolution = Resolution.builder()
+                .providerUrl(NamingServiceType.CNS, BuildConfig.BLOCKCHAIN_NET_URL + BuildConfig.INFURA_API_KEY)
+                .build();
+        address = library.getAddress(domain, "eth");
+        return address
+    }
+}
+
+class UnstoppableDomainsRepository() {
+    suspend fun resolve(domain: String): Solidity.Address {
+        val futureTask: FutureTask<String> = FutureTask<String>(BackgroundTask(domain))
```

Comment:
please remove this future task. This function is suspending, so it is possible to wait for the resolution using async / await.
```kotlin
val address = async {  library.getAddress(domain, "eth") }.await()
```

Reply:
I have done it and Android Studio suggested using Kotlin's `withContext(Dispatchers.IO) {}` instead of async / await.
I believe it acts similarly to async { ... }.await()